keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -64,22 +162,26 @@ class SqlVerificationRepository(
   private inline fun <reified RESULT> ResultQuery<*>.fetchOneInto() =
     fetchOneInto(RESULT::class.java)
 
+  /**
+   * Defaults a timestamp field to [EPOCH] if it is null.
+   */
+  private fun Field<LocalDateTime>.orEpoch(): Field<LocalDateTime> =
+    isnull(this, EPOCH.atZone(UTC).toLocalDateTime())
```

Comment:
This is what makes the left join to the "last checked" table work. If the timestamp is `null` (because there is no corresponding row in the last checked table) then we get the default `EPOCH` meaning anything that was never checked will order first and definitely be older than the cutoff.